### DISCUSSION TITLE
ROSディストリビューションがコンテナの中に入っていない

### DISCUSSION THREAD

**Discussion by ShutoTakashita** [Tiryoh/docker-ros2-desktop-vnc#69] [Q&A] [answered]
先日「Unityで始める　ROS　人工知能　ロボットプログラミング」を読み始めたのですが、
docker run -p 6080:80 --shm-size=1024m tiryoh/ros2-desktop-vnc:galactic
を実行し、起動されたコンテナ内のターミナルでrosversionと入力しても command not foundと出てきてしまいます。
調べていてsupervisorのconfigファイル？を書き換えれば良いのかとも思ったのですが、手順のどこでインストールしたのかもわからず、困っています。何が悪いのでしょうか？

github上のdockerfileからビルドしたものでも同じエラーが出ました。
また、dockerfile内でsupervisorをインストールしている箇所が見つからないのに上記のエラーが出ていることも分かりません。

![image](https://user-images.githubusercontent.com/50667299/191662613-23f6b3ab-179e-4ad6-a669-8a78fc068414.png)

**Comment by ShutoTakashita** [comment, marked as answer]
失礼しました、ROS2でバージョンを調べるコマンドはrosversionではありませんでした。

また、勘違いから本の著者さん=このレポジトリの作成者だと思ってここで質問してしまいました。申し訳ございません。

**Reply by Tiryoh** [reply]
こんばんは。npakaさんの書籍にて紹介していただいたそうですが、私はまだ中身を読めていないので`rosversion`コマンドについてのみ記載します。

`rosversion`コマンドは`python3-rospkg`をインストールすると使用できるようです（このDisucussionにてはじめてその存在を認知しました）
https://command-not-found.com/python3-rosversion

インストールすればROS 1/ROS 2かかわらずROSのディストリビューションを確認することができました。

この`python3-rospkg`の依存関係を調べてみたところ、ROS 1をインストールした環境ではインストールされるようです。ROS 2の環境でもTerminalを起動して`python3-rospkg`をインストールすれば`rosversion`コマンドが使用できるようになると思います。

![Screenshot from 2022-09-23 01-09-51](https://user-images.githubusercontent.com/3256629/191798362-994273a0-371a-49e2-9199-3ce2a9ad3c2a.png)

**Reply by Tiryoh** [reply]
それと、supervisorについてはこのリポジトリのDockerfileでベースにしている[dorowu/ubuntu-desktop-lxde-vnc](https://github.com/fcwu/docker-ubuntu-vnc-desktop)にてインストールされています。
https://github.com/fcwu/docker-ubuntu-vnc-desktop/blob/e4922ce92f945fc482994b7a0fd95ca5de7295b3/Dockerfile.amd64#L20

足りないパッケージを最初からインストールしておきたいということであれば、supervisorの設定をいじるよりも、こちらのDockerfileを編集してビルドしてみることをおすすめします。
https://github.com/Tiryoh/docker-ros2-desktop-vnc/blob/master/galactic/Dockerfile

**Reply by ShutoTakashita** [reply]
ご丁寧にありがとうございます。無事、python3-rospkgをインストールしてコマンドが使えることを確認しました。また、supervisorの方についてもご返信くださりありがとうございます。
